### PR TITLE
Fixes #9726: Improve error messages in "rudder agent update"

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/9726

### CODE REVIEW

**Comment by amousset** [comment]
Still untested, do not merge for now.

**Comment by ncharles** [comment]
This is so awesome :)

File: techniques/system/common/1.0/update.st [hunk 1]
```diff
@@ -129,11 +129,23 @@ bundle agent update
...
       "*********************************************************************************
 * rudder-agent could not get an updated configuration from the policy server.   *
 * This can be caused by a network issue, an unavailable server, or if this      *
-* node was deleted from the Rudder root server.                                 *
+* node's IP in not if the allowed networks of its policy server.                *
+* Any existing configuration policy will continue to be applied without change. *
+*********************************************************************************"
+      action => immediate;
+
+     !rudder_ncf_hash_update_error.(rudder_promises_generated_error|no_update)::
```
Comment (ncharles) [inline techniques/system/common/1.0/update.st]: it should be rudder_promises_generated_tmp_file_error

**Comment by amousset** [comment]
Commit modified

File: techniques/system/common/1.0/update.st (hunk 1 above)
Comment (amousset) [inline techniques/system/common/1.0/update.st]: Only since 3.2 I think.

**Comment by amousset** [comment]
@ncharles Ready for review, this has been tested.

File: techniques/system/common/1.0/update.st (hunk 1 above)
Comment (ncharles) [inline techniques/system/common/1.0/update.st]: indeed

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
OK, merging this PR

**Pull request merged**